starter-code/iteration-3/main.ts

```diff
@@ -1,8 +1,56 @@
-// Add the reference to the interface
+import {TodoItemInterface, TodoListInterface} from './interfaces';
 
-// Create class TodoItem that implements the corresponding interface
+class TodoItem implements TodoItemInterface{
+  title:string;
+  done:boolean = false;
+  updatedAt:Date;
 
-// Create class TodoList that implements the corresponding interface
+  constructor(title:string){
+    this.title = title;
+    this.updatedAt = new Date();
+  }
+
+  toggleStatus(){
+    this.done = !this.done;
+    this.updatedAt = new Date();
+  }
+}
+
+class TodoList implements TodoListInterface{
+  todos:Array<TodoItemInterface> = [];
+
+  addTask(item:TodoItemInterface){
+    this.todos.push(item);
+    console.log("Task inserted in the list");
+    return this.todos.length;
+  }
+  
+  listAllTasks():void{
+    for(var i =0; i<this.todos.length; i++){
+      console.log(this.todos[i].title);
+    }
+  }
+
+  listUncomplete() {
+    this.todos.forEach( (item) => {
+      if (!item.done){
+        console.log(item.title);
```

Comment:
you could improve this with interpolation:
`console.log(`Pending: ${task.title}, ${task.updatedAt}`);`